firewood/src/db.rs

```diff
@@ -339,6 +339,13 @@ pub struct Proposal<'p> {
     db: &'p Db,
 }
 
+impl Proposal<'_> {
+    /// Get the root hash of the proposal synchronously
+    pub fn root_hash_sync(&self) -> Result<Option<api::HashKey>, api::Error> {
+        self.nodestore.root_hash().map_err(api::Error::from)
```

Comment:
Idiomatic, since ? automatically calls `Into::into`
```suggestion
        Ok(self.nodestore.root_hash()?)
```